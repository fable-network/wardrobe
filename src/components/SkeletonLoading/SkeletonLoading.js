import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { theme } from '../../theme';
import { skeleton, fadeIn } from '../../animations';

const defaultCss = css`
  display: block;
  min-height: 1px;
  width: ${(p) => p.width || '100%'};
  background: ${p => p.baseColor || p.theme.grey05};
  position: relative;
  &:before {
    content: '';
    ${(p) => (p.animating && `animation: ${skeleton(p.color)} 2.5s ease-in-out infinite;`)}
  }
`;

const textCss = css`
  height: 16px;
`;

const squareCss = css`
  padding-top: ${(p) => p.width || '100%'};
`;

const headerCss = css`
  height: 24px;
`;

const circleCss = css`
  ${squareCss};
  border-radius: 50%;
  overflow: hidden;
`;

const Wrapper = styled.div`
  ${defaultCss};
  ${(p) => p.height && `height: ${p.height};`}
  ${(p) => p.shape === 'text' && textCss};
  ${(p) => p.shape === 'square' && squareCss};
  ${(p) => p.shape === 'header' && headerCss};
  ${(p) => p.shape === 'circle' && circleCss};
  animation: ${fadeIn} 500ms ease;
`;

const SkeletonLoading = ({ width, animating, shape, color, height, baseColor }) => (
  <Wrapper
    width={width}
    animating={animating}
    shape={shape}
    color={color}
    height={height}
    baseColor={baseColor}
  />
);

SkeletonLoading.propTypes = {
  /** Used to set the width of the skeleton block; preferably this should be left at 100% and
  the width set by the parent */
  width: PropTypes.string,
  /** Use the height prop to define custom height that is not provided by one of the shapes */
  height: PropTypes.string,
  animating: PropTypes.bool,
  shape: PropTypes.oneOf(['text', 'square', 'header', 'circle']),
  /** color used for the animation */
  color: PropTypes.string,
  /** color used for the base */
  baseColor: PropTypes.string,
};

SkeletonLoading.defaultProps = {
  width: '100%',
  animating: true,
  baseColor: theme.grey05,
};

export default SkeletonLoading;
