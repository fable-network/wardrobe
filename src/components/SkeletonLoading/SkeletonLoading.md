Header Loading:

```jsx
<SkeletonLoading shape="header" />
```

Text Loading:

```jsx
<SkeletonLoading shape="text" />
```

Square Loading:

```jsx
<SkeletonLoading width="200px" shape="square" />
```

Circle Loading:

```jsx
<SkeletonLoading width="200px" shape="circle" />
```

No shape (custom width and height):

```jsx
<SkeletonLoading width="50%" height="100px" />
```

Custom animation color:

```jsx
<SkeletonLoading color="cyan" shape="text" />
```

Custom base color:

```jsx
<SkeletonLoading baseColor="#333" shape="text" />
```

Without animation:

```jsx
<SkeletonLoading animating={false} shape="header" />
```

Real world examples:

- Profile card:

```jsx
<div
  style={{
    display: 'flex',
    alignItems: 'center',
    boxShadow: '0 1px 4px rgba(0,0,0,0.2)',
    padding: '16px',
    width: '400px',
    maxWidth: '100%',
  }}
>
  <div style={{ marginRight: '16px' }}>
    <SkeletonLoading shape="circle" width="100px" />
  </div>
  <div style={{ width: '100%' }}>
    <div style={{ marginBottom: '8px' }}>
      <SkeletonLoading shape="header" width="100%" />
    </div>
    <div style={{ marginBottom: '8px' }}>
      <SkeletonLoading shape="text" width="70%" />
    </div>
    <SkeletonLoading shape="text" width="60%" />
  </div>
</div>
```

- Table with header:

```jsx
<div
  style={{
    boxShadow: '0 1px 4px rgba(0,0,0,0.2)',
    padding: '16px',
    width: '600px',
    maxWidth: '100%',
  }}
>
  <div
    style={{ display: 'flex', justifyContent: 'space-between', width: '70%', marginBottom: '24px' }}
  >
    <SkeletonLoading shape="text" width="150px" />
    <SkeletonLoading shape="text" width="150px" />
  </div>
  <div style={{ marginBottom: '16px' }}>
    <SkeletonLoading animating={false} height="32px" />
  </div>
  <div style={{ marginBottom: '16px' }}>
    <SkeletonLoading shape="text" />
  </div>
  <SkeletonLoading shape="text" />
</div>
```
