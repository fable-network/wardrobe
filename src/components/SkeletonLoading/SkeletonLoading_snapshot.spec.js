/* eslint-disable */
// This file was automatically generated
import React from 'react';
import 'jest-styled-components';
import SkeletonLoading from './SkeletonLoading';


jest.mock('../../charts/HighChart', () => () => <high-charts />);

describe('SkeletonLoading Component', () => {
  describe('snapshots', () => {
    it('should match snapshot', () => {
    
      const wrapper1 = mount(
        <SkeletonLoading shape="header" />
      );
      expect(wrapper1).toMatchSnapshot();

      const wrapper2 = mount(
        <SkeletonLoading shape="text" />
      );
      expect(wrapper2).toMatchSnapshot();

      const wrapper3 = mount(
        <SkeletonLoading width="200px" shape="square" />
      );
      expect(wrapper3).toMatchSnapshot();

      const wrapper4 = mount(
        <SkeletonLoading width="200px" shape="circle" />
      );
      expect(wrapper4).toMatchSnapshot();

      const wrapper5 = mount(
        <SkeletonLoading width="50%" height="100px" />
      );
      expect(wrapper5).toMatchSnapshot();

      const wrapper6 = mount(
        <SkeletonLoading color="cyan" shape="text" />
      );
      expect(wrapper6).toMatchSnapshot();

      const wrapper7 = mount(
        <SkeletonLoading baseColor="#333" shape="text" />
      );
      expect(wrapper7).toMatchSnapshot();

      const wrapper8 = mount(
        <SkeletonLoading animating={false} shape="header" />
      );
      expect(wrapper8).toMatchSnapshot();

    });
  });
});
