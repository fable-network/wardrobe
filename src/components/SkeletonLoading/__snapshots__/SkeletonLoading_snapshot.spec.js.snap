// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonLoading Component snapshots should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SkeletonLoading
    animating={true}
    baseColor="#e4e8ea"
    shape="header"
    width="100%"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animating": true,
      "baseColor": "#e4e8ea",
      "color": undefined,
      "height": undefined,
      "shape": "header",
      "width": "100%",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animating": true,
        "baseColor": "#e4e8ea",
        "color": undefined,
        "height": undefined,
        "shape": "header",
        "width": "100%",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`SkeletonLoading Component snapshots should match snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SkeletonLoading
    animating={true}
    baseColor="#e4e8ea"
    shape="text"
    width="100%"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animating": true,
      "baseColor": "#e4e8ea",
      "color": undefined,
      "height": undefined,
      "shape": "text",
      "width": "100%",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animating": true,
        "baseColor": "#e4e8ea",
        "color": undefined,
        "height": undefined,
        "shape": "text",
        "width": "100%",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`SkeletonLoading Component snapshots should match snapshot 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SkeletonLoading
    animating={true}
    baseColor="#e4e8ea"
    shape="square"
    width="200px"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animating": true,
      "baseColor": "#e4e8ea",
      "color": undefined,
      "height": undefined,
      "shape": "square",
      "width": "200px",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animating": true,
        "baseColor": "#e4e8ea",
        "color": undefined,
        "height": undefined,
        "shape": "square",
        "width": "200px",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`SkeletonLoading Component snapshots should match snapshot 4`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SkeletonLoading
    animating={true}
    baseColor="#e4e8ea"
    shape="circle"
    width="200px"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animating": true,
      "baseColor": "#e4e8ea",
      "color": undefined,
      "height": undefined,
      "shape": "circle",
      "width": "200px",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animating": true,
        "baseColor": "#e4e8ea",
        "color": undefined,
        "height": undefined,
        "shape": "circle",
        "width": "200px",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`SkeletonLoading Component snapshots should match snapshot 5`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SkeletonLoading
    animating={true}
    baseColor="#e4e8ea"
    height="100px"
    width="50%"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animating": true,
      "baseColor": "#e4e8ea",
      "color": undefined,
      "height": "100px",
      "shape": undefined,
      "width": "50%",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animating": true,
        "baseColor": "#e4e8ea",
        "color": undefined,
        "height": "100px",
        "shape": undefined,
        "width": "50%",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`SkeletonLoading Component snapshots should match snapshot 6`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SkeletonLoading
    animating={true}
    baseColor="#e4e8ea"
    color="cyan"
    shape="text"
    width="100%"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animating": true,
      "baseColor": "#e4e8ea",
      "color": "cyan",
      "height": undefined,
      "shape": "text",
      "width": "100%",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animating": true,
        "baseColor": "#e4e8ea",
        "color": "cyan",
        "height": undefined,
        "shape": "text",
        "width": "100%",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`SkeletonLoading Component snapshots should match snapshot 7`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SkeletonLoading
    animating={true}
    baseColor="#333"
    shape="text"
    width="100%"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animating": true,
      "baseColor": "#333",
      "color": undefined,
      "height": undefined,
      "shape": "text",
      "width": "100%",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animating": true,
        "baseColor": "#333",
        "color": undefined,
        "height": undefined,
        "shape": "text",
        "width": "100%",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`SkeletonLoading Component snapshots should match snapshot 8`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SkeletonLoading
    animating={false}
    baseColor="#e4e8ea"
    shape="header"
    width="100%"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animating": false,
      "baseColor": "#e4e8ea",
      "color": undefined,
      "height": undefined,
      "shape": "header",
      "width": "100%",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animating": false,
        "baseColor": "#e4e8ea",
        "color": undefined,
        "height": undefined,
        "shape": "header",
        "width": "100%",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
